playbooks/common/openshift-master/restart_hosts.yml

```diff
@@ -7,14 +7,27 @@
   ignore_errors: true
   become: yes
 
+# WARNING: This process is riddled with weird behavior.
+#
+# Ansible's blog documents this *without* the port, which appears to now
+# just wait until the timeout value and then proceed without checking anything.
+# port is now required.
+#
+# However neither ansible_ssh_port or ansible_port are reliably defined, likely
+# only if overridden. Assume a default of 22.
+#
+# Using local_action causes ansible_host to be set to "localhost". The bug below
+# documents a workaround but this variable does not appear to exist anymore.
+# For now it appears this must be 'inventory_hostname'. (as it was previously)
+# See: https://github.com/ansible/ansible/issues/16139
 - name: Wait for master to restart
   local_action:
     module: wait_for
-      host="{{ ansible_host }}"
+      host="{{ inventory_hostname }}"
```

Comment:
arrgh, right...  We can't rely on openshift.common.hostname, since we might not have access to the internal network...  openshift.common.public_hostname/openshift.common.public_ip isn't much more reliable there either...

Can we save the values of ansible_port and ansible_host from a run that isn't local_action?  Does this affect `delegate_to: localhost` as well?